storage/rocksdb/ha_rocksdb.cc

```
@@ -3697,6 +3715,12 @@ static int rocksdb_init_func(void* const p)
     mysql_mutex_unlock(&rdb_sysvars_mutex);
   }
 
+  if (rocksdb_persistent_cache_size > 0) {
```

Should also verify the path is valid or fail bringing up the server.

---

I think it is best to let rocksdb handle that, no?

---

To expand, it is also no trivial to determine if the cache size is valid so to save messy logic of user specifying one option and not the other (and I don't think its wise to check that user specified both size and path because then the cache could fail silently) I use the size as the flag, send values in and let rdb validate everything.

---

Ah, okay. RocksDB checks to see if the path is empty and fails the setup. That would be fine. It wasn't clear what would happen when the empty string was passed in and if RocksDB would create a default one.

---

Yes, that is fine since it looks like RocksDB checks for empty string and invalidates. It wasn't clear what the default behavior was.